generators/browser-compatible-base/src/ast/AbstractWriter.ts

```diff
@@ -150,6 +150,24 @@ export class AbstractWriter {
         this.indentLevel--;
     }
 
+    public delimit<T extends AbstractAstNode>(
+        nodes: T[],
+        delimiter: string,
+        writeFunction: (node: T) => void,
+        prefix: string = "",
+        suffix: string = ""
+    ): void {
+        this.write(prefix);
+        if (nodes.length > 0) {
+            writeFunction(nodes[0]!);
```

Comment:
We might need to refactor things a bit to prevent any need for the `!` operator.

What if we did a length check then iterated one by one? We can also start by always using newlines to separate arguments (otherwise lines could get huge).

```typescript
if (nodes.length === 0) {
  this.write(prefix);
  this.write(suffix);
  return;
}
this.writeLine(prefix);
for (const node of nodes) {
  writeFunction(node);
  this.writeLine(delimiter);
}
this.writeLine(suffix);
```

Reply:
True, I think the benefit of parameterizing the delimiter is that we could plumb this through to a config eventually for how a given customer wants to format it